rfcs/45-stageless.md

```diff
@@ -0,0 +1,736 @@
+(Note to readers: this is not the same as "system chaining" in Bevy 0.6 and earlier: that concept has been renamed to "system piping".)
+
+```rust
+fn main(){
+   App::new()
+   // These systems are connected using a string of ordering constraints
+
+   .add_systems((compute_attack, 
+                 compute_defense,
+                 check_for_crits,
+                 compute_damage,
+                 deal_damage,
+                 check_for_death)
+                 .chain()
+                 // We can configure all systems in the chain at once
+                 .to(GameSet::Combat)
+   )
+   .run()
+}
+```
+
+#### Run criteria
+
+While ordering constraints determine *when* a system will run, **run criteria** will determine *if* it will run at all.
+Each system can have any number of run criteria, which read data from the world to control whether or not that system runs.
+If (and only if) all of its run criteria return `true`, the system will run.
+If any of the run criteria are `false`, the system will be skipped.
+Systems that are skipped are considered completed for the purposes of ordering constraints.
+
+Run criteria are not systems, but can be created from systems which can read (but not write) data from the `World` and returns a boolean value.
+
+You can specify run criteria in several different ways:
+
+```rust
+// This is just an ordinary system: timers need to be ticked!
+fn tick_construction_timer(timer: ResMut<ConstructionTimer>, time: Res<Time>){
+    timer.tick(time.delta());
+}
+
+// This function can be used as a run criterion system,
+// because it does not mutate data and returns `bool`
+fn construction_timer_finished(timer: Res<ConstructionTimer>) -> bool {
+    timer.finished()
+}
+
+// You can use queries, event readers and resources with arbitrarily complex internal logic
+fn too_many_enemies(population_query: Query<(), With<Enemy>>, population_limit: Res<PopulationLimit>) -> bool {
+   population_query.iter().count() > population_limit
+}
+
+fn main(){
+    App::new()
+    .add_plugins(DefaultPlugins)
+    // We can add functions with read-only system parameters as run criteria
+    .add_system((
+       tick_construction_timer, 
+       update_construction_progress)
+       .chain()
+       .run_if(construction_timer_finished)
+    )
+   .add_system(system_meltdown_mitigation.run_if(too_many_enemies))
+    // We can use closures for simple one-off run criteria, 
+    // which automatically fetch the appropriate data from the `World`
+    .add_system(spawn_more_enemies.run_if(|difficulty: Res<Difficulty>| difficulty >= 9000))
+    // The `run_if_resource_equals` method is fast, special-cased syntax that generates a run criterion
+    // for when you want to check the value of a resource (commonly an enum)
+    .add_system(gravity.run_if_resource_equals(Gravity::Enabled))
+    // Run criteria can be attached to system sets
+    .add_set(GameSet::Physics.run_if_resource_equals(Paused(false)))

```

Comment:
What happens if I don't call this for a label? What if I call it multiple times? Maybe `configure_set` would fit better here?

Also see my main comment: `configure_label`?

Reply:
Implemented, good suggestion.